core/controllers/simulationcontroller/simulation_controller.go

```diff
@@ -54,6 +57,33 @@ func (c *SimulationController) WithPrefix(prefix string) *SimulationController {
 	}
 }
 
+func (c *SimulationController) EndSimulation(rw http.ResponseWriter, r *http.Request) {
+	id := mux.Vars(r)["id"]
+	if id == "" {
+		rw.WriteHeader(http.StatusBadRequest)
+		rw.Write([]byte(`{"Err":"Simulation id cannot be empty"}`))
+		return
+	}
+
+	intVar, err := strconv.Atoi(id)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
```

Comment:
Same here I would make this a 422 Unprocessable Entity instead of a 400 Bad Request